spec.html

```diff
@@ -46724,6 +46897,83 @@ <h1>Proxy.revocable ( _target_, _handler_ )</h1>
     </emu-clause>
   </emu-clause>
 
+  <emu-clause id="sec-module-source-objects">
+    <h1>Module Source Objects</h1>
+    <p>Module Source Objects represent modules in their source import phase, which are not linked, instantiated or executed.</p>
+    <p>All Module Source Objects should have a prototype of %AbstractModuleSource%.prototype and must define a [[ModuleSourceRecord]] internal slot.</p>
+    <p>Hosts may define their own Module Source subclasses for custom module record types.</p>
+
+    <emu-clause id="sec-%abstractmodulesource%-constructor">
+      <h1>The %AbstractModuleSource% Constructor</h1>
+      <p>The %AbstractModuleSource% constructor:</p>
+      <ul>
+        <li>is <dfn>%AbstractModuleSource%</dfn>.</li>
+        <li>along with its corresponding prototype object, provides common properties that are inherited by module source constructors and their instances.</li>
+        <li>does not have a global name or appear as a property of the global object.</li>
+        <li>acts as the abstract superclass of the _ModuleSource_ constructor.</li>
```

Comment:
```suggestion
        <li>acts as a common superclass of the constructors of Module Source Objects.</li>
```

I'm not 100% sure about this one, but I think it's better.